L72-core-google_default_credentials-extension.md

```diff
@@ -0,0 +1,65 @@
+Allow Call Credentials to be Specified in `grpc_google_default_credentials_create`
+----
+* Author(s): rbellevi
+* Approver: markdroth
+* Status: Draft
+* Implemented in: Core
+* Last updated: July 8th, 2020
+* Discussion at: https://groups.google.com/g/grpc-io/c/fZNm4pU8e3s/m/2Be8u1n7BQAJ
+
+## Abstract
+
+This document proposes that `grpc_google_default_credentials_create` be
+amended to allow the user to specify their desired call credentials.
+
+## Background
+
+The Google default credentials created by the
+`grpc_google_default_credentials_create` function in Core enable connection to
+Google services via a combination of ALTS and SSL credentials, along with a special oauth2
+token that must assert the same identity as the channel-level ALTS credential.
```

Comment:
lets note here that the identity we're discussing here is specifically the one authenticated by the token available on the metadata server's `http://metadata.google.internal/computeMetadata/v1/project/service-accounts/default/token` endpoint.

Reply:
sorry, to be clear I meant to say that we should clarify that the ALTS channel-level identity is that authenticated by the token served from this metadata server endpoint.

Google default credentials picks it's per-RPC metadata token via the ADC spec, though (which by default is, but in general isn't necessarily going to result in a per-RPC token from that same endpoint).

Reply:
You're right. The current wording is ambiguous. Updated to clarify that it's the ALTS identity that's taken from that HTTP endpoint.